### PR TITLE
use npm-sass to find bin

### PR DESCRIPTION
**Pull request by JoeChapman**
Use npm-sass to find install path of node-sass

### CODE REVIEW

File: bin/tasks.js
Comment (lennym) [inline bin/tasks.js]: This doesn't seem right. You're effectively `toString`-ing the module to get the string `npm-sass`. Just use the string.

**Review by lennym — CHANGES_REQUESTED** [review]

File: bin/tasks.js
Comment (joefitter) [inline bin/tasks.js]: What are you doing here? isn't npmsass the actual module? Are these my hands?

Comment (JoeChapman) [reply bin/tasks.js]: yep, my bad

Comment (lennym) [reply bin/tasks.js]: In fact, since this isn't an npm script, you'll need the fully qualified path to the binary, so I'd leave as it was with `witch` finding the binary path.

File: bin/tasks.js
Comment (joefitter) [inline bin/tasks.js]: do we need to include this at all?

**Review by joefitter — CHANGES_REQUESTED** [review]

File: bin/tasks.js
Comment (joefitter) [reply bin/tasks.js]: we should either use npmsass programmatically or just use the name of the binary

**Review by sulthan-ahmed — APPROVED** [review]

File: bin/tasks.js
Comment (joefitter) [reply bin/tasks.js]: personally, I would prefer all of this to be in just plain ol boring JS code, include the modules, use them, etcetera. But I know you love your dirty bash

Comment (sulthan-ahmed) [reply bin/tasks.js]: o yeah I don't think you need this.  I did it without it

**Review by sulthan-ahmed — CHANGES_REQUESTED** [review]
remove `const npmsass = require('npm-sass');`

**Comment by JoeChapman** [comment]
@lennym @joefitter @sulthan-ahmed changes made

**Review by joefitter — APPROVED** [review]

File: bin/tasks.js
Comment (lennym) [inline bin/tasks.js:12]: You need to `witch` this, same as before.

**Comment by joefitter** [comment]
actually @JoeChapman why are we eslintignoring bin? That dupe var would have been picked up if we linted it

File: bin/tasks.js
Comment (lennym) [reply bin/tasks.js:12]: Actually, there's an outside chance you *might* not need to, npm might work it out since you're inside the `bin` from package.json, but I'd want to check that first.

`witch` will be safe though.

File: bin/tasks.js
Comment (lennym) [reply bin/tasks.js]: > personally, I would prefer all of this to be in just plain ol boring JS code

I'm inclined to agree, not least because there's absolutely zero error handling code in this setup at the moment. If things fail you get a nice console.error, but the process doesn't actually fail (i.e. exits with 0)

**Comment by JoeChapman** [comment]
@joefitter will address the eslintignore stuff in separate pr

**Comment by joefitter** [comment]
> > personally, I would prefer all of this to be in just plain ol boring JS code

> I'm inclined to agree, not least because there's absolutely zero error handling code in this setup at the moment. If things fail you get a nice console.error, but the process doesn't actually fail (i.e. exits with 0)

I'll raise an issue for this - unless you guys would prefer it to be addressed before merging?

**Comment by lennym** [comment]
@joefitter Makes sense.

No reason at all IMO to block merge - it's the same on master, it's just something that I spotted when I was reviewing/trying to work out what was going on.

**Comment by JoeChapman** [comment]
@joefitter I'm not sure I follow, what is the context of that comment?
You would rather the eslint amends were made in this PR?
Or you want changes to the binary, i.e. proper error handling.. in this PR or another?

**Comment by joefitter** [comment]
@JoeChapman rather than doing:
```json
"browserify": "browserify ./target ./output --plugin ./blah"
```

we would do:

```js
var browserify = require('browserify');
var b = browserify();
b.add('./browser/main.js');
b.bundle().pipe(process.stdout);
```

**Comment by JoeChapman** [comment]
ok, fair enough. However, I would prefer not to change all the things in this PR. Can you raise an issue so we can discuss...?
In the interim, I will fix the linting.
If this PR, minus those issues, is satisfactory please do me a solid

**Comment by joefitter** [comment]
@JoeChapman already did

**Comment by joefitter** [comment]
https://github.com/UKHomeOfficeForms/hof-frontend-assets/issues/4

**Comment by JoeChapman** [comment]
I need one more +1 @lennym @sulthan-ahmed @easternbloc @gxxm

File: bin/tasks.js
Comment (lennym) [reply bin/tasks.js:12]: This needs to be `${npmsass}`

**Comment by lennym** [comment]
NOOOO! https://github.com/UKHomeOfficeForms/hof-frontend-assets/pull/3/files#r94939370

**Pull request merged**
